tests/testthat/test-metadata.R

```diff
@@ -7,122 +6,518 @@ extract_format <- function(.x) {
   df <- data.frame(x = "a", y = "b")
-  varmeta <- data.frame(dataset  = rep("df", 2),
-                    variable = c("x", "y"),
-                    label    = c("foo", "Lorem ipsum dolor sit amet, consectetur adipiscing elit"))
-  dfmeta <- data.frame(dataset  = "df",
-                   label = "Lorem ipsum dolor sit amet, consectetur adipiscing elit")
-  
-  expect_warning(xportr_label(df, varmeta),
-               "variable label must be 40 characters or less")
-  expect_error(xportr_df_label(df, dfmeta),
-               "dataset label must be 40 characters or less")
+  df_meta <- data.frame(dataset = "df", label = "Label")
+
+  df_spec_labeled_df <- xportr_df_label(df, df_meta)
+
+  expect_equal(attr(df_spec_labeled_df, "label"), "Label")
+  expect_equal(
+    dput(df_spec_labeled_df),
+    structure(list(x = "a", y = "b"), class = "data.frame", row.names = c(NA, -1L), label = "Label")
+  )
+})
+
+test_that("xportr_df_label: Correctly applies label when data is piped", {
+  df <- data.frame(x = "a", y = "b")
+  df_meta <- data.frame(dataset = "df", label = "Label")
+
+  df_spec_labeled_df <- df %>% xportr_df_label(df_meta)
+
+  expect_equal(attr(df_spec_labeled_df, "label"), "Label")
+  expect_equal(
+    dput(df_spec_labeled_df),
+    structure(
+      list(x = "a", y = "b"),
+      class = "data.frame", row.names = c(NA, -1L), `_xportr.df_arg_` = "df", label = "Label"
+    )
+  )
+})
+
+test_that("xportr_df_label: Correctly applies label for custom domain", {
+  df <- data.frame(x = "a", y = "b")
+  df_meta <- data.frame(dataset = "DOMAIN", label = "Label")
+
+  df_spec_labeled_df <- xportr_df_label(df, df_meta, domain = "DOMAIN")
+
+  expect_equal(attr(df_spec_labeled_df, "label"), "Label")
+  expect_equal(
+    dput(df_spec_labeled_df),
+    structure(
+      list(x = "a", y = "b"),
+      class = "data.frame", row.names = c(NA, -1L), `_xportr.df_arg_` = "DOMAIN", label = "Label"
+    )
+  )
+})
+
+test_that("xportr_df_label: Correctly applies label for metacore spec", {
+  skip_if_not_installed("metacore")
+  df <- data.frame(x = "a", y = "b")
+  metacore_meta <- suppressWarnings(
+    metacore::metacore(
+      ds_spec = data.frame(
+        dataset = c("df"),
+        structure = "",
+        label = c("Label")
+      )
+    )
+  )
+
+  metacore_spec_labeled_df <- xportr_df_label(df, metacore_meta)
+
+  expect_equal(attr(metacore_spec_labeled_df, "label"), "Label")
+  expect_equal(
+    dput(metacore_spec_labeled_df),
+    structure(
+      list(x = "a", y = "b"),
+      class = "data.frame",
+      row.names = c(NA, -1L), label = "Label"
+    )
+  )
+})
+
+test_that("xportr_df_label: Expect error if label exceeds 40 characters", {
+  df <- data.frame(x = "a", y = "b")
+  df_meta <- data.frame(
+    dataset = "df",
+    label = strrep("a", 41)
+  )
+
+  expect_error(
+    xportr_df_label(df, df_meta),
+    "dataset label must be 40 characters or less"
+  )
+})
```

Comment:
I think these are duplicates of the above tests?

Reply:
As discussed on the call, Pretty much the only thing that changed is the function we're testing.